src/sql/base/browser/ui/table/table.ts

```diff
@@ -140,11 +145,12 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 		this.mapMouseEvent(this._grid.onDblClick, this._onDoubleClick);
 		this._grid.onColumnsResized.subscribe(() => this._onColumnResize.fire());
 		this._grid.onRendered.subscribe(() => {
-			if (!this._grid.getActiveCell()) {
-				this._container.tabIndex = this._data.getLength() > 0 ? 0 : -1;
-			}
+			// When the grid is rerendered (e.g. view switching), the active cell information is kept but the cell will become not keyboard focusable (tabindex changed from 0 to -1).
+			// we need to set the container's tabindex to 0 to go through the focus redirection logic.
+			this._container.tabIndex = this._container.querySelector('[tabindex = "0"]') || this._data.getLength() === 0 ? -1 : 0;
```

Comment:
Are you trying to look at the current tabindex of the container or purposely looking at all children too?

Reply:
let me update the comments.